python/sparkdl/param/converters.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=wrong-spelling-in-docstring,invalid-name,import-error
+
+""" SparkDLTypeConverters
+Type conversion utilities for definition Spark Deep Learning related MLlib `Params`.
+"""
+
+import six
+
+import tensorflow as tf
+
+from pyspark.ml.param import TypeConverters
+
+import sparkdl.utils.keras_model as kmutil
+
+__all__ = ['SparkDLTypeConverters']
+
+class SparkDLTypeConverters(object):
+    """
+    .. note:: DeveloperApi
+
+    Factory methods for type conversion functions for :py:func:`Param.typeConverter`.
+    These methods are similar to :py:class:`spark.ml.param.TypeConverters`.
+    They provide support for the `Params` types introduced in Spark Deep Learning Pipelines.
+    """
+
+    @staticmethod
+    def toTFGraph(value):
+        """
+        Convert a value to a :py:obj:`tf.Graph` object, if possible.
+        """
+        if not isinstance(value, tf.Graph):
+            raise TypeError("Could not convert %s to tf.Graph" % type(value))
+        return value
+
+    @staticmethod
+    def asColumnToTensorNameMap(value):
+        """
+        Convert a value to a column name to :py:obj:`tf.Tensor` name mapping
+        as a sorted list of string pairs, if possible.
+        """
+        if not isinstance(value, dict):
+            err_msg = "Could not convert [type {}] {} to column name to tf.Tensor name mapping"
+            raise TypeError(err_msg.format(type(value), value))
+
+        # Conversion logic after quick type check
+        strs_pair_seq = []
+        for _maybe_col_name, _maybe_tnsr_name in value.items():
+            # Check if the non-tensor value is of string type
```

Comment:
the function name is self-explanatory so you don't need this comment. comments aren't alway good because the code logic can change and the comment can be forgotten and not changed. there are lots of wrong comments in many codebases.